hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java

```diff
@@ -601,7 +590,28 @@ private void purgeZeroAdjustments(TransferList.Builder xfers) {
 		} while (lastZeroRemoved != -1);
 	}
 
-	public boolean isKnownTreasury(AccountID aId) {
-		return tokenStore.isKnownTreasury(aId);
+	private ResponseCodeEnum tryTokenChange(BalanceChange change) {
+		var validity = OK;
+		var tokenId = tokenStore.resolve(change.tokenId());
+		if (tokenId == MISSING_TOKEN) {
+			validity = INVALID_TOKEN_ID;
+		}
+		if (validity == OK) {
+			validity = adjustTokenBalance(change.accountId(), tokenId, change.units());
+			if (validity == INSUFFICIENT_TOKEN_BALANCE) {
+				validity = change.codeForInsufficientBalance();
+			}
+		}
+		return validity;
+	}
+
+	private void adjustHbarUnchecked(List<BalanceChange> changes) {
+		for (var change : changes) {
+			if (change.isForHbar())	{
+				final var accountId = change.accountId();
+				setBalance(accountId, change.getNewBalance());
+				updateXfers(accountId, change.units(), netTransfers);
+			}
+		}
```

Comment:
nit : new methods that are validating `BalanceChange` like adjustHbarUnchecked, tryTokenChange, plausibilityOf can be moved to a new utils class ?

Reply:
Agreed! Moved `tryTokenChange(BalanceChange)` to be a `default` method on the `TokenStore`.